text/trace/0000-error_flagging.md

```diff
@@ -0,0 +1,79 @@
+
+### Status Codes
+Currently, OpenTelemetry does not have a use case for differentiating between different types of errors. However, this use case may appear in the future. For now, we would like to reduce the number of status codes, and then add them back in as the need becomes clear. We would also like to differentiate between status codes which have not been
+set, and an explicit OK status set by an end user.
+
+* `UNSET` is the default status code.
+* `ERROR` represents all error types.
+* `OK` represents a span which has been explicitly marked as being free of errors, and should not be counted against an error budget. Note that only end users should set this status. Instead, instrumentation should leave the status as `UNSET` for nominal operations.
+
+### `Status Source`
+A new Status Source field identifies the origin of the status code on the span. This is important, as statuses set by application developers and operators have been confirmed by the end user to be correct to the particular situation. Statuses set by instrumentation, on the other hand, are only following a generic schema.
+
+* `INSTRUMENTATION` is the default source. This is used for instrumentation contained within shared code, such as OSS libraries and frameworks. All instrumentation plugins shipped with OpenTelemetry use this status code.
+* `APPLICATION` identifies statuses set by the application developer, within their application code. 
+* `OPERATOR` identifies statuses which have been altered during data egress. 
+
+Analysis tools MAY disregard status codes, in favor of their own approach to error analysis. However, it is strongly suggested that analysis tools SHOULD pay attention to the status codes when set by `APPLICATION` or `OPERATOR`, as it is a communication from the end-user and contains valuable information.
+ 
+### Status Mapping Schema
+As part of the specification, OpenTelemetry provides a canonical mapping of semantic conventions to status codes. This removes any ambiguity as to what OpenTelemetry ships with out of the box. 
+
+Please note that semantic conventions, and thus status mapping from conventions, are still a work in progress and will continue to change after GA.
+ 
+### Status Processor
+The collector will provide a processor and a configuration language to make adjustments to this status mapping schema. This provides the flexibility and customization needed for real world scenarios.
+ 
+### Convenience methods
+As a convenience, OpenTelemetry provides helper functions for adding semantic conventions and exceptions to a span. These helper functions will also set the correct status code. This simplifies the life of the instrumentation author, and helps ensure compliance and data quality.
+
+Note that these convenience methods simply wire together multiple API calls. They should live in a helper package, and should not be directly added to existing API interfaces. Given how many semantic conventions we have, there will be a pile of them.
```

Comment:
Are these helper functions part of API or SDK or neither? What's the compatibility guarantee?
Is this something that each language SDK will decide, or the spec will cover it (for consistency)?

Reply:
These helper functions should be higher order functions which make a series of API calls on your behalf. They could go in their own package, but either way they should not be added directly to the Span API.

Reply:
Should we also move RecordException out of the core span interface API? I guess this is a bit of a special case because vendors could have large differences in what data they record here.

Reply:
I think we should keep that as it simply creates an `Event` (and should not affect directly the final `Status` value).

Reply:
But doensn't this OTEP want to change that?

> As a convenience, OpenTelemetry provides helper functions for adding semantic conventions and exceptions to a span. These helper functions will also set the correct status code.

Reply:
Although I concur that exceptions should not affect the status by default (unless they are escaped #784)

Reply:
I think `RecordException()` is low-level enough to keep it IMHO (unless @tedsuo thinks otherwise).

Reply:
Either way; I'm fine with keeping RecordException as low level, for the reasons @Oberon00 mentions. I imagine that instrumentation authors will have way more use of these functions than application developers, so keeping RecordException as part of the Span API would be helpful as it is very important.

Rather than add yet-another-package, these functions could be added to the same package that contains the constants for semantic conventions. That may be the best place for them.